resolwe/flow/executors/__init__.py

```diff
@@ -122,6 +123,14 @@ def run(self, data_id, script, verbosity=1):
 
         self.data_id = data_id
 
+        # Fetch data instance to get any executor requirements.
+        if data_id != 'no_data_id':
```

Comment:
I don't remember exactly, but I think that this hack was introduced to support "get or create", where we don't want to process new objects in standard way. However I don't now if this is still used.